mobx_codegen/lib/src/type_names.dart

```diff
@@ -23,53 +23,63 @@ import 'package:mobx_codegen/src/template/comma_list.dart';
 class LibraryScopedNameFinder {
   LibraryScopedNameFinder(this.library);
 
-  final LibraryElement library;
+  final LibraryElement2 library;
 
-  final Map<Element, String?> _namesByElement = {};
+  final Map<Element2, String?> _namesByElement = {};
 
-  Map<Element, String?> get namesByElement {
+  Map<Element2, String?> get namesByElement {
     // Add all of this library's type-defining elements to the name map
-    final libraryElements =
-        library.topLevelElements.whereType<TypeDefiningElement>();
+    final libraryElements = library.children2.whereType<TypeDefiningElement2>();
     for (final element in libraryElements) {
-      _namesByElement[element] = element.name;
+      _namesByElement[element] = element.name3;
     }
 
     // Reverse each import's export namespace so we can map elements to their
-    // library-local names. Note that the definedNames include a prefix if there
-    // is one.
-
-    for (final import in library.definingCompilationUnit.libraryImports) {
-      for (final entry in import.namespace.definedNames.entries) {
-        _namesByElement[entry.value] = entry.key;
+    // library-local names.
+    //
+    // Prior to analyzer 7.4.0, definedNames included the import prefix, but it
+    // no longer does with the new element API. This means we need to construct
+    // it manually.
+
+    final libraryImports = library.fragments
+        .map((fragment) => fragment.libraryImports2)
+        .expand((t) => t);
+
+    for (final import in libraryImports) {
+      final prefix = import.prefix2;
+      for (final entry in import.namespace.definedNames2.entries) {
+        _namesByElement[entry.value] = prefix?.name2 != null
+            // If the prefix exists, we prepend it to the name
+            ? '${prefix!.name2!}.${entry.key}'
+            // Otherwise, we just use the name
+            : entry.key;
```

Comment:
A note on this change. The old code appeared to rely on `definedNames` including prefix. So in this case:

```dart
import 'dart:io' as io;
```

... all the names from `definedNames` would already include the prefix, as in `io.File`, `io.Directory`, and so on. This is no longer the case, so I am instead constructing the full element name manually.